pkg/controller/clusterpool/collections.go

```diff
@@ -0,0 +1,548 @@
+	// age, oldest first.
+	Unassigned() []*hivev1.ClusterClaim
+	// ByName returns the named claim from the lookup, or nil if no claim by that name exists.
+	ByName(string) *hivev1.ClusterClaim
+	// SyncCDAssignments makes sure each claim which purports to be assigned has the correct CD
+	// assigned to it, updating the CD and/or claim on the server as necessary.
+	SyncCDAssignments(client.Client, cdLookup, log.FieldLogger)
+	// Assign assigns the specified claim to the specified cluster, updating its spec and status on
+	// the server. Errors updating the spec or status are bubbled up. Returns an error if the claim is
+	// already assigned (to *any* CD). Does *not* validate that the CD isn't already assigned (to this
+	// or another claim).
+	Assign(client.Client, *hivev1.ClusterClaim, *hivev1.ClusterDeployment) error
+	// Untrack removes the named claims from the claimLookup, so they are no longer
+	// - counted when doing pool math
+	// - considered for assignment
+	// Do this to broken claims.
+	Untrack(...string)
+}
+
+type cdLookup interface {
+	// ByName returns the named ClusterDeployment from the cdLookup, or nil if no CD by that name exists.
+	ByName(string) *hivev1.ClusterDeployment
+	// Installing returns the list of ClusterDeployments in the process of being installed. These are
+	// not available for claim assignment.
+	Installing() []*hivev1.ClusterDeployment
+	// Assignable returns a list of ClusterDeployment refs, sorted according to which we feel ought to
+	// be assigned first to minimize claim response time.
+	// - Running CDs, oldest first
+	// - Resuming CDs, in order of least recently resumed (soonest likely to become Running)
+	// - Hibernating CDs, oldest first
+	// - Anything else, oldest first. (These should probably be skipped. TODO?)
+	Assignable() []*hivev1.ClusterDeployment
+	// Deleting returns the list of ClusterDeployments whose DeletionTimestamp is set. Not to be
+	// confused with MarkedForDeletion.
+	Deleting() []*hivev1.ClusterDeployment
+	// MarkedForDeletion returns the list of ClusterDeployments with the
+	// ClusterClaimRemoveClusterAnnotation. Not to be confused with Deleting: if a CD has its
+	// DeletionTimestamp set, it is *not* included in MarkedForDeletion.
+	MarkedForDeletion() []*hivev1.ClusterDeployment
+	// Total returns the total number of ClusterDeployments in the cdLookup.
+	Total() int
+	// Assign assigns the specified ClusterDeployment to the specified claim, updating its spec on the
+	// server. Errors from the update are bubbled up. Returns an error if the CD is already assigned
+	// (to *any* claim).
+	Assign(client.Client, *hivev1.ClusterDeployment, *hivev1.ClusterClaim) error
+	// Invalidate idempotently removes the named ClusterDeployments from the assignable list of the
+	// cdLookup, so they are no longer considered for assignment. They still count against pool
+	// capacity. Do this to a broken ClusterDeployment.
+	Invalidate(...string)
```

Comment:
this name does not clarify use, and we have a very specific use of removing them from being assigned. would like to see the name reflect that.

Reply:
How about `MakeNotAssignable`?

Reply:
that sounds better to me 👍🏽